modules/EmailMan/EmailMan.php

```diff
@@ -1003,6 +1003,179 @@ public function mark_deleted($id)
+            } else {
+                $log->error('Incorrect bean');
+                return false;
+            }
+        }
+
+        $focus = BeanFactory::getBean($type, $id);
+        if(!$focus) {
+            throw new Exception('Email address has not related bean.');
+        }
+
+        $ret = $this->sendOptInEmailViaMailer($focus, $emailAddress);
+        return $ret;
+    }
+
+    /**
+     *
+     * @global LoggerManager $log
+     * @global array $sugar_config
+     * @param SugarBean $focus
+     * @param EmailAddress $emailAddress
+     * @return boolean
+     */
+    private function sendOptInEmailViaMailer(
+        SugarBean $focus,
+        EmailAddress $emailAddress
+    ) {
+
+        global $log,  $sugar_config;
+
+        $confirmOptInEnabled = isset($sugar_config['email_enable_confirm_opt_in']) && $sugar_config['email_enable_confirm_opt_in'];
+
+        if (!$confirmOptInEnabled) {
+            $this->warn('Confirm Opt In disabled');
+            return false;
+        }
+
+        $ret = true;
+
+        $emailTemplate = new EmailTemplate();
+
+        $confirmOptInTemplateId =
+                isset($sugar_config['aop']['confirm_opt_in_template_id']) ?
+                $sugar_config['aop']['confirm_opt_in_template_id'] : null;
+        if(!$confirmOptInTemplateId) {
+            $configurator = new Configurator();
+            $confirmOptInTemplateId =
+                $configurator->config['email_confirm_opt_in_email_template_id'];
+        }
+
+        if (!$confirmOptInTemplateId) {
+            $log->fatal(
+                'Opt In Email Template is not configured.'
+                . ' Please set up in email settings'
+            );
+            SugarApplication::appendErrorMessage(
+                    $app_strings['ERR_OPT_IN_TPL_NOT_SET']);
+            return false;
+        }
+
+        $emailTemplate->retrieve($confirmOptInTemplateId);
+
+        $mailer = new SugarPHPMailer();
+        $mailer->setMailerForSystem();
+
+        $emailObj = new Email();
+        $defaults = $emailObj->getSystemDefaultEmail();
+
+        $mailer->From = $defaults['email'];
+        $mailer->FromName = $defaults['name'];
+
+        $mailer->Subject = from_html($emailTemplate->subject);
+
+        $mailer->Body = from_html($emailTemplate->body_html);
+        $mailer->Body_html = from_html($emailTemplate->body_html);
+        $mailer->AltBody = from_html($emailTemplate->body);
+
+        if (is_string($emailAddress->email_address)) {
+            $emailAddressString = $emailAddress->email_address;
+        } elseif (is_array($emailAddress->email_address) && is_string($emailAddress->email_address[0]['email_address'])) {
+            $emailAddressString = $emailAddress->email_address[0]['email_address'];
+        } else {
+            $log->fatal('Incorrect Email Address');
+            return false;
+        }
+
+        $mailer->addAddress($emailAddressString, $focus->name);
+
+        $mailer->replace('user_first_name',
```

Comment:
finally we eliminate the $user_first_name, $user_last_name etc.. we should remove this part then.